modules/emulator/R/minimize.GP.R

```diff
@@ -211,33 +212,45 @@ mcmc.GP <- function(gp, x0, nmcmc, rng, format = "lin", mix = "joint", splinefcn
   for (g in start:nmcmc) {
     
     if (mix == "joint") {
-      ## propose new
-      xnew <- xcurr
+
+      # adapt
       if ((g > 2) && ((g - 1) %% settings$assim.batch$jump$adapt == 0)) {
         params.recent <- samp[(g - settings$assim.batch$jump$adapt):(g - 1), ]
         colnames(params.recent) <- names(x0)
         # accept.count <- round(jmp@arate[(g-1)/settings$assim.batch$jump$adapt]*100)
         jcov <- pda.adjust.jumps.bs(settings, jcov, accept.count, params.recent)
         accept.count <- 0  # Reset counter
       }
+      
+      ## propose new
       repeat {
         xnew <- mvrnorm(1, unlist(xcurr), jcov)
         if (bounded(xnew, rng)) {
           break
         }
       }
       # if(bounded(xnew,rng)){
-      currY <- get.y(gp, xcurr, n.of.obs, llik.fn, priors, settings)
+      
+      # update model parameters conditioned on likelihood parameters
+      # when llik.par (= currll) is passed get.y returns posterior value
+      currY <- get.y(gp, xcurr, n.of.obs, llik.fn, priors, settings, currll)
       ycurr <- currY$posterior.prob
-      pcurr <- currY$par
-      newY  <- get.y(gp, xnew, n.of.obs, llik.fn, priors, settings, currY$llik.par)
+
+      newY  <- get.y(gp, xnew, n.of.obs, llik.fn, priors, settings, currll)
       ynew  <- newY$posterior.prob
+
       if (is.accepted(ycurr, ynew)) {
         xcurr <- xnew
-        pcurr <- newY$par
-        
+        #pcurr <- newY$par
+        #curll <- newY$llik.par
```

Comment:
In this version, you're only updating ```xcurr <- xnew``` but you also need to update ```SScurr <- SSnew``` (though I can't tell in the code where you're tracking the current summary statistics here, is that ycurr and ynew?)